### DISCUSSION TITLE
How can i let window in screen centered

### DISCUSSION THREAD

**Discussion by USTHzhanglu** [hecrj/iced#2011] [Q&A]
I want make my window in screen centered like qt:
``` c++
MainWindow w;
int currentScreen = a.desktop()->screenNumber(&w);//程序所在的屏幕编号
QRect rect = QGuiApplication::screens().at(currentScreen)->geometry();//程序所在屏幕尺寸
w.move((rect.width() - w.width()) / 2, (rect.height() - w.height()) / 2);//移动到所在屏幕中间
```

and i find api [`iced::window::fetch_size`](https://docs.iced.rs/iced/window/fn.fetch_size.html) may like this. but I don't know how to use it.
and it seems get iced window size but my desktop size.
does iced support get screen size?

**Comment by i509VCB** [comment]
Somewhat related note: on some platforms (Wayland in particular), this isn't actually possible.

**Comment by USTHzhanglu** [comment]
i find it.
use:
```
        window: iced::window::Settings {
            position: iced::window::Position::Centered,
            ..iced::window::Settings::default()
        },
```
